### ISSUE TITLE
Bring back old behavior of UnsafeCast

### ISSUE THREAD

**Issue by xouabita** [arturbosch/detekt#2131]
After https://github.com/arturbosch/detekt/pull/1879, `UnsafeCast` have the same behavior as the kotlin compiler warning `CAST_NEVER_SUCCEEDS`:

code:
```kotlin
fun main() {
    val foo = "bar"
    println(foo as Int)
}
```

command:
```bash
kotlinc test.kt -include-runtime
```

output:
```bash
test.kt:3:17: warning: this cast can never succeed
    println(foo as Int)
```

It makes this rule redundant if you already have warnings as errors. 

It would be nice to bring back the [old behavior of this rule](https://github.com/arturbosch/detekt/blob/399e7f6128e6e6ace2400585ca36dbddb7246e24/detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/UnsafeCast.kt#L38), behind a config or as a new rule.

See also: https://github.com/arturbosch/detekt/issues/2082#issuecomment-557034738

**Comment by schalkms** [comment]
I'm against reverting back to the old behavior.
I would favor to stick and comply with IntelliJ's rule behavior.

**Comment by arturbosch** [comment]
@schalkms not even through a config flag? I mean by default we use the same behavior as IntelliJ but the user can choose if he wants all `as` expressions being reported.

**Comment by schalkms** [comment]
I don't think so. Whilst I favor configurable behavior in rules, I'm against making the complete rule logic configurable. Essentially, we would have two rules in one. Hence, it's harder to maintain.
I remember that we received complains and issues about the old behavior of this rule. Here we adhere to the standard IDEA logic, which is nice.
If the OP really wants to have the old behavior back (remember it has some flaws), he can copy the code from the history to a new [custom rule](https://arturbosch.github.io/detekt/extensions.html#custom-rulesets).

**Comment by 3flex** [comment]
I agree with @schalkms, I don't think it makes sense to have a config option for this, for the reasons he listed and also because I don't think we should set a precedent of creating a config option that's known to report false positives.

**Comment by xouabita** [comment]
Hi, thank you all for your answers 👍 
Which false positives did the old behavior have ?
I thought the old behavior was too aggressive for some users but it would be nice to have it back for the users who really want it.

**Comment by schalkms** [comment]
I don't even know if I caught all of them, but here are the ones that I found while doing a quick search.

- #726 
- #939 
- #1601
- #1137
